### PR TITLE
SI-5833 Fixes tail-of-Nil problem in RefinedType#normalizeImpl

### PR DESCRIPTION
**Pull request by JamesIry**
RefinedType#normalizeImpl was checking to see if the flattened list of
parents had an empty tail then pulling the head if so. But if the list
was empty then boom. This fix makes it check if the whole list has
length 1 instead. Empty lists will flow through to the rest the logic
which has no problems with Nil.

review @adriaanm or @hubertp

Questions for reviewer to contemplate:
I had no luck finding a reproducing test case. Can you?
Should a refinement type with no parents and no decls normalize to AnyRef instead of falling through to the rest of the logic?

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Types.scala [hunk 1]
```diff
@@ -1800,7 +1800,7 @@ trait Types extends api.Types { self: SymbolTable =>
       // TODO see comments around def intersectionType and def merge
       def flatten(tps: List[Type]): List[Type] = tps flatMap { case RefinedType(parents, ds) if ds.isEmpty => flatten(parents) case tp => List(tp) }
       val flattened = flatten(parents).distinct
-      if (decls.isEmpty && flattened.tail.isEmpty) {
+      if (decls.isEmpty && (flattened lengthCompare 1 == 0)) {
```
Comment (retronym) [inline src/reflect/scala/reflect/internal/Types.scala]: There is a convenience for that in the complier:

`hasLength(flattened, 1)`

**Comment by JamesIry** [comment]
How embarrassing. I'm blaming the Eclipse plugin.

File: src/reflect/scala/reflect/internal/Types.scala (hunk 1 above)
Comment (JamesIry) [inline src/reflect/scala/reflect/internal/Types.scala]: Also has the advantage that I don't have to remember how precedence works. :-)

**Comment by adriaanm** [comment]
LGTM

**Pull request merged**